content/blog/2023-12-11-arrays-and-variable-arguments-in-Bril.md

```diff
@@ -0,0 +1,96 @@
++++
+title = "Variable Arguments in Bril and Arrays in the TypeScript compiler"
+[extra]
+bio = """
+ Alice is an M.Eng student studying Computer Science. She is broadly interested in compilers, systems and algorithm design.
+"""
+[[extra.authors]]
+name = "Alice Sze"
++++
+
+# What was the goal?
+
+The overall goal of my project is to improve the capabilities and the user experience of Bril. In particular, I set out to allow the `main` function in Bril to take a variable number of arguments (much like `argc` and `argv` in C/C++), and add support for arrays in the TypeScript compiler to Bril. I thought of adding variable arguments because we currently have to pack a fixed number of values into an array when writing Bril programs, which is not only extra effort but also limits us to testing a very specific set of inputs, e.g. an array of six elements. While it is not a huge modification, it does limit our ability to test our programs and to see the effect of our optimizations on larger inputs. I also included arrays in the TypeScript compiler because quite a few benchmarks are written using it, and I recall someone asking if arrays are supported for the first assignment. I hope that these extensions will make it ever so slightly easier for CS6120ers to write interesting programs in the future.
```

Comment:
I recommend using a "," after occurrences of "e.g." and "i.e.". https://capra.cs.cornell.edu/styleguide/#egie